src/int/mul.rs

```diff
@@ -0,0 +1,181 @@
+//! [`Int`] multiplication operations.
+
+use core::ops::{Mul, MulAssign};
+
+use subtle::{Choice, ConstantTimeEq, CtOption};
+
+use crate::{Checked, CheckedMul, Int, Uint, Zero};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// Compute "wide" multiplication as a 3-tuple `(lo, hi, sgn)`.
+    /// The `(lo, hi)` components contain the _magnitude of the product_, with sizes
+    /// correspond to the sizes of the operands; the `sgn` indicates the sign of the result.
+    pub fn split_mul<const RHS_LIMBS: usize>(
```

Comment:
it would be nice to have it const fn

Reply:
Fixed it.

Reply:
we already have `Uint::select`

Reply:
Well, guess I'm blind! 
Let me know what you think of this new solution.